batch/batch/worker/worker.py

```diff
@@ -1676,38 +1678,42 @@ async def run(self):
 
                 log.info(f'{self}: running jvm process')
                 with self.step('running'):
-                    self.process = await asyncio.create_subprocess_exec(
-                        *self.command_string,
-                        stdout=asyncio.subprocess.PIPE,
-                        stderr=asyncio.subprocess.PIPE,
-                        env={envvar['name']: envvar['value'] for envvar in self.env},
+                    await self.jvm.execute(
+                        f'{local_jar_location}', self.user_command_string[0], self.user_command_string[1:]
                     )
-
-                    await asyncio.gather(self.pipe_to_log(self.process.stdout), self.pipe_to_log(self.process.stderr))
-                    await self.process.wait()
-
-                log.info(f'finished {self} with return code {self.process.returncode}')
-
-                await self.worker.file_store.write_log_file(
-                    self.format_version, self.batch_id, self.job_id, self.attempt_id, 'main', self.logbuffer.decode()
-                )
-
-                if self.process.returncode == 0:
-                    self.state = 'succeeded'
-                else:
-                    self.state = 'failed'
+                self.state = 'succeeded'
                 log.info(f'{self} main: {self.state}')
             except asyncio.CancelledError:
                 raise
             except Exception:
+                # FIXME: this can also be a Hail Query driver error, not a Hail Batch error
                 log.exception(f'while running {self}')
 
                 self.state = 'error'
                 self.error = traceback.format_exc()
                 await self.cleanup()
-            await self.cleanup()
+            else:
```

Comment:
Again, let's see what the logic is for making sure logs get uploaded when the job can be cancelled.